src/Raven.Server/ServerWide/ServerStore.cs

```diff
@@ -174,8 +177,20 @@ public ServerStore(RavenConfiguration configuration, RavenServer server)
                            Uri.TryCreate(Configuration.Core.ServerUrls[0], UriKind.Absolute, out var uri) == false ||
                            uri.Port != 0;
 
-            var maxConcurrentBackupsTasks = Configuration.Backup.MaxNumberOfConcurrentBackups ?? int.MaxValue;
-            ConcurrentBackupsSemaphore = new SemaphoreSlim(maxConcurrentBackupsTasks);
+            _skipChangingMaxConcurrentBackups = Configuration.Backup.MaxNumberOfConcurrentBackups != null;
+            GlobalBackupConfiguration = new GlobalBackupConfiguration(Configuration.Backup.MaxNumberOfConcurrentBackups ?? int.MaxValue);
+        }
+
+        public void UpdateMaxConcurrentBackups(int licensedCores)
+        {
+            if (_skipChangingMaxConcurrentBackups)
+                return;
+
+            var maxNumberOfConcurrentBackups = Math.Max(licensedCores / 2, 1);
+            if (maxNumberOfConcurrentBackups == GlobalBackupConfiguration.MaxNumberOfConcurrentBackups)
+                return;
+
+            GlobalBackupConfiguration = new GlobalBackupConfiguration(maxNumberOfConcurrentBackups);
```

Comment:
You are creating a new instance here, but you already passed the instance to other locations.
That means that we'll have two lock objs.

Reply:
This is until the backup finishes.

Reply:
Done